decoders/svg/src/main/java/com/squareup/picasso3/decoder/svg/SvgImageDecoder.java

```diff
@@ -0,0 +1,52 @@
+package com.squareup.picasso3.decoder.svg;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import com.caverock.androidsvg.SVG;
+import com.caverock.androidsvg.SVGParseException;
+import com.squareup.picasso3.ImageDecoder;
+import com.squareup.picasso3.Request;
+import java.io.IOException;
+import okio.BufferedSource;
+
+class SvgImageDecoder implements ImageDecoder {
+
+  @Override public boolean canHandleSource(BufferedSource source) {
+    try {
+      SVG.getFromInputStream(source.peek().inputStream());
```

Comment:
is it sufficient to check for xml header bytes?

Reply:
The library internally handles gzipped content here, so we'd have to move that logic out into this, and we'd also have to validate the SVG.  It could be done but I think that should live in the SVG library.

Reply:
That's a fair point.  My point was simply that determining if the stream is an SVG is a little more complex that check that it (1) is an XML file and (2) has a DOCTYPE or something.  Taking out the GZipInputStream case (which I guess is for http streams), the existing SVG library will validate the SVG as well as load it, which is something that makes sense in an SVG parser, as opposed to reimplementing here.

One example of the added complexity, from what I can see, the existing library checks the DOCTYPE, which might contain `<!ENTITY` in which case it falls back to a different parser.  I think understanding the complexities around XML, and SVG in particular, make sense to be handled in the parser.

Ideally the library had another method that would simply check if the stream is an SVG without parsing the whole thing, but this one doesn't.